### ISSUE TITLE
Dome light texture is ignored

### ISSUE THREAD

**Issue by gbottesi** [PixarAnimationStudios/USD#1708]
### Description of Issue
I am using Storm renderer with a HdxTaskController with a UsdImagingDelegate.
If I create a dome light and set it an environment texture, calling setDomeLightTextureFile, the parameter is simply ignored by task controller which always read the default value.

In taskController.cpp, the following code seems to be responsible:

```
if (light.IsDomeLight()) {
    _delegate.SetParameter(_lightIds[i], 
                        HdLightTokens->textureResource, 
                        _defaultDomeLightTextureResource);
    _delegate.SetParameter(
        _lightIds[i], HdLightTokens->textureFile,
        SdfAssetPath(
            HdxPackageDefaultDomeLightTexture(),
            HdxPackageDefaultDomeLightTexture()));
}
```

### System Information (OS, Hardware)
Windows, OpenGL renderer

**Comment by jilliene** [comment]
Filed as internal issue #USD-7060

**Comment by florianhecht** [comment]
I've run into the same issue and fixed it in my clone of the repo. I had to fix two locations:

1. In `HdxTaskController::_SetParameters`
2. In `HdxTaskController::_SetBuiltInLightingState`

After fixing those two locations I ran into another issue. After the dome light has been created, changes to its transformation are ignored (unless the light is removed and then added back). I traced the bug to `HdxTaskController::_SetBuiltInLightingState` where in the loop over lights to check for updates the transformation is not extracted again and the transformation of the SPrim is not marked dirty.

https://github.com/PixarAnimationStudios/USD/blob/release/pxr/imaging/hdx/taskController.cpp#L794

I fixed it by inserting this at the top
```
_delegate.SetParameter(_lightIds[i], 
                       HdTokens->transform,
                       lights[i].GetTransform());
```
and then marking the transform dirty as well.
```
GetRenderIndex()->GetChangeTracker().MarkSprimDirty(
        _lightIds[i], HdLight::DirtyTransform | HdLight::DirtyParams);
```
I hope this helps fix two issues in one go.

Cheers, Florian
